guide/popular-topics/permissions.md

```diff
@@ -0,0 +1,335 @@
+
+To check your bots permissions in the channel the command was used in you could use something like this:
+
+```js
+// Final permissions for a guild member using permissionsFor
+const botPermissionsFor = channel.permissionsFor(guild.me);
+
+// Final permissions for a guild member using permissionsIn
+const botPermissionsIn = guild.me.permissionsIn(channel);
+
+// Final permissions for a role
+const rolePermissions = channel.permissionsFor(myRole);
+```
+
+<warning>The `.permissionsFor()` method returns a bit field with all permissions set if the member or role has the global `ADMINISTRATOR` permission and does not take overwrites into consideration in this case. Using the second parameter of the `.has()` method as described further down in the guide will not allow you to check without taking `ADMINISTRATOR` into account here!</warning>
+
+If you want to know how to work with the returned Permissions objects keep reading as this will be our next topic.
+
+## The Permissions object
+
+The [Permissions object](https://discord.js.org/#/docs/main/stable/class/Permissions) is a Discord.js class containing a permissions bit field and a bunch of utility methods to manipulate it easily.
+Remember that using these methods will not manipulate permissions, but create a new instance representing the changed bit field.
+
+### Converting permission numbers
+
+Some methods and properties in Discord.js return permission decimals rather than a Permissions object, making it hard to manipulate or read them if you don't want to use bitwise operations.
+You can however pass these decimals to the Permissions constructor to convert them.
+
+```js
+const { Permissions } = require('discord.js');
+
+// constructing a Permissions Object from a permission number
+const permissions = new Permissions(268550160);
+// constructing a Permissions Object from an array of PermissionResolvables
+const flags = ['MANAGE_CHANNELS', 'EMBED_LINKS', 'ATTACH_FILES', 'READ_MESSAGE_HISTORY', 'MANAGE_ROLES'];
+const permissions2 = new Permissions(flags);
+```
+
+### Checking for permissions
+
+The Permissions object features the `.has()` method allowing an easy way to check flags in a Permissions bit field.
+The `.has()` method takes two parameters. The first being either a permission number, single flag or an array of permission numbers and flags, the second being a boolean (true or false) indicating if you want to allow the `ADMINISTRATOR` permission to override (defaults to true).
+
+Let's say you want to know if the decimal bit field representation `268550160` has `MANAGE_CHANNELS` referenced:
+
+```js
+const { Permissions } = require('discord.js');
+
+const permissions = new Permissions(268550160);
+console.log(permissions.has('MANAGE_CHANNELS'));
+// output: true
```

Comment:
These comments can go inline, e.g.:
```js
const permissions = new Permissions(268550160);
console.log(permissions.has('MANAGE_CHANNELS')); // true
```

Though, I think we have an eslint rule against this; not entirely sure. If so, just space them out instead.